nextjs-app/utils/index.ts

```diff
@@ -15,3 +15,11 @@ export const expandArray = <T>(array: T[]): T[] => {
   }
   return result;
 };
+
+export const shuffleArray = <T>(array: T[]): T[] => {
+  for (let i = array.length - 1; i > 0; i--) {
+    const j = Math.floor(Math.random() * (i + 1));
+    [array[i], array[j]] = [array[j], array[i]];
+  }
+  return array;
+};
```

Comment:
I noticed that the current `shuffleArray` implementation mutates the input array directly, which could lead to unexpected side effects. I would suggest creating a copy of the array before shuffling to make it a pure function: